be/src/storage/rowset/vectorized/segment_iterator.h

```diff
@@ -5,15 +5,214 @@
 #include <memory>
 #include <vector>
 
+#include "storage/del_vector.h"
+#include "storage/rowset/column_iterator.h"
+#include "storage/rowset/segment.h"
+#include "storage/rowset/vectorized/segment_options.h"
 #include "storage/vectorized/chunk_iterator.h"
 
-class Segment;
+namespace starrocks {
+class ColumnDecoder;
+}
 
 namespace starrocks::vectorized {
 
-class ColumnPredicate;
-class Schema;
-class SegmentReadOptions;
+/// SegmentIterator
+// TODO(zhuming): Refine the implementation of this class to reduce the intellectual overhead.
+// Too many policies encapsulated in this class, should split this class into many small classes.
+class SegmentIterator final : public ChunkIterator {
```

Comment:
Why move this to the header file?

Reply:
use column_iterator directly, instead segment iterator.